### PR TITLE
Tests and improvements for Utils::Misc

### PR DESCRIPTION
**Pull request by Waqar144**

### CODE REVIEW

File: src/utils/misc.cpp [hunk 1]
```diff
@@ -961,13 +964,13 @@ int Utils::Misc::getDefaultSearchEngineId() {
...
  */
 QList<int> Utils::Misc::getSearchEnginesIds() {
-    QList<int> list;
-    list << SearchEngines::DuckDuckGo << SearchEngines::Google
-         << SearchEngines::Bing << SearchEngines::Yahoo
-         << SearchEngines::GoogleScholar << SearchEngines::Yandex
-         << SearchEngines::AskDotCom << SearchEngines::Qwant
-         << SearchEngines::Startpage;
-    return list;
+    return QList<int> {
```
Comment (pbek) [inline src/utils/misc.cpp]: Will this work with older Qt versions?

File: QOwnNotes-Project.pro [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+SUBDIRS = \
+    src \
+    tests
+
+src.file = src/QOwnNotes.pro
+tests.file = tests/QOwnNotesTests.pro
+
+test.depends = src
+
```
Comment (pbek) [inline QOwnNotes-Project.pro]: Why the name `QOwnNotes-Project.pro`? Wouldn't `QOwnNotes.pro` be better?

Comment (Waqar144) [reply QOwnNotes-Project.pro]: Just to differentiate from the inner QOwnNotes.pro file :grimacing:

File: src/utils/misc.cpp (hunk 1 above)
Comment (Waqar144) [reply src/utils/misc.cpp]: Yeah totally.

File: QOwnNotes-Project.pro (hunk 2 above)
Comment (Waqar144) [reply QOwnNotes-Project.pro]: Renamed inner files to `src.pro` and `tests.pro`. Outer file is now `QOwnNotes.pro`

Comment (pbek) [reply QOwnNotes-Project.pro]: But I still don't like that `QOwnNotes-Project.pro`. Why do we need that again?

Comment (Waqar144) [reply QOwnNotes-Project.pro]: Just for utility. Tests and Project are opened as one overall project with two subprojects(app, tests) so you can quickly switch from one to another

Comment (pbek) [reply QOwnNotes-Project.pro]: Isn't that almost more confusing than before? Are there other projects who are using one project file for the application and the tests?

Comment (Waqar144) [reply QOwnNotes-Project.pro]: Yeah most of them

But I will remove it.

Comment (pbek) [reply QOwnNotes-Project.pro]: I'm not against it if it is considered "best practice". :laughing: And that was what I tried to find out.

Comment (Waqar144) [reply QOwnNotes-Project.pro]: I don't know if its best practice lol :rofl: , but it does kinda make it easier to run the tests(from Qt creator, i.e., just switch the target)

**Comment by Waqar144** [comment]
We just hit 5000 commits :fireworks:

File: QOwnNotes-Project.pro (hunk 2 above)
Comment (pbek) [reply QOwnNotes-Project.pro]: I just switch project.

**Comment by pbek** [comment]
Let's have a commit party. :tada: :laughing:

**Pull request merged**